docs/index.md

```diff
@@ -45,3 +45,156 @@ or the WCS API (which will also install `wcs-lib`):
+      "Who's there?";
+      "Who is there?";
+      "Who are you?";
+    ]
+    ()
+```
+
+So, we can know define the next node in the conversation that is going
+to respond to the _who is there?_ question:
+
+```ocaml
+let whoisthere =
+  Wcs.dialog_node "WhoIsThere"
+    ~conditions_spel: (Spel.intent who_intent)
+    ~text: "Broken Pencil"
+    ~parent: knock
+    ()
+```
+
+We can notice that here the condition is not given as a string but as
+an expression written using the embedding of the Spel expression
+language (used by WCS) inside OCaml.
+
+In the rest of the conversation the user is expected to repeat the
+name of the character given by the bot. To be sure that the user
+repeat the name, we are going to define an entity `char_entity`
+containing the name or some synonyms:
+
+```ocaml
+let char_entity =
+  Wcs.entity "Characters"
+    ~values: [ "Broken Pencil", ["Dammaged Pen"; "Fractured Pencil"] ]
+    ()
+```
+
+If the input given by the user contains the name of the character, then
+the bot terminates the joke:
+
+```ocaml
+let answer =
+  Wcs.dialog_node "Answer"
+    ~conditions_spel: (Spel.entity char_entity ())
+    ~text: "Nevermind it's pointless"
+    ~parent: whoisthere
+    ~context: (Context.return (Json.bool true))
+    ()
+```
+
+Setting a `return` field in the context will case the termination of
+the bot. If the user doesn't gives the name of the character, we want
+to tell the user what to do. So we add a fallback node:
+
+```ocaml
+let fallback =
+  Wcs.dialog_node "Fallback"
+    ~conditions_spel: Spel.anything_else
+    ~text: "You should repreat my name!"
+    ~previous_sibling: answer
+    ~next_step: (whoisthere, Wcs_aux.Goto_body)
+    ()
+```
+
+We can now build the entire workspace containing all the dialog nodes,
+entities, and intents:
+
+```ocaml
+let ws_knockknock =
+  Wcs.workspace "Knock Knock"
+    ~entities: [ char_entity ]
+    ~intents: [ who_intent ]
+    ~dialog_nodes: [ knock; whoisthere; answer; fallback; ]
+    ()
+```
+
+It is possible to print this workspace:
+
+```ocaml
+let () = print_endline (Wcs_pretty.workspace ws_knockknock)
+```
+
+But it is also possible to deploy the workspace on WCS directly from
+OCaml. For that, the library needs the service credentials:
+
+```ocaml
+let wcs_cred = Wcs_bot.get_credential None
+```
+
+The call to this function [`Wcs_bot.get_credential`](XXX TODO XXX) is
+going lookup in the environment variable `$WCS_CRED` to find a path to
+a file name containing the service credentials in the following format:
+
+```js
+{
+  "url": "https://gateway.watsonplatform.net/conversation/api",
+  "password": "PASSWORD",
+  "username": "USERNAME"
+}
+```
+
+Now, with the service credentials, we can deploy the workspace on WCS:
+
+```ocaml
+let create_rsp = Wcs_api.create_workspace wcs_cred ws_knockknock
+```
+
+Finally, we can try the bot with the function [`Wcs_bot.exec`](XXX TODO XXX)
+ and providing the workspace identifier that has just been created:
+
```

Comment:
Finally, we can try the bot with the function [`Wcs_bot.exec`](XXX TODO XXX)
providing the workspace identifier that has just been created: